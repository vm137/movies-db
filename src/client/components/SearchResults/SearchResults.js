import React, {Component} from 'react';
import './SearchResults.scss';


export default class SearchResults extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {
        return (
          <div className="results-wrapper">
            <h2>search results</h2>
            <div>{this.props.data}</div>
          </div>
        );
    }
}
