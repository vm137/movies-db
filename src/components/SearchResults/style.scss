.results-wrapper {
  overflow: auto;
  margin-top: 20px;
}
