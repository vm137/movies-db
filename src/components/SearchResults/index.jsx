import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SearchResults from './SearchResults';

SearchResults.propTypes = {
  moviesR: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = state => ({
  moviesR: state.movies,
});

export default connect(mapStateToProps)(SearchResults);
