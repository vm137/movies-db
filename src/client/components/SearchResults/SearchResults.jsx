import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import MovieCard from '../MovieCard';
import './style.scss';

const SearchResults = ({ moviesR }) => (
  <div className="results-wrapper">
    {(moviesR).map(movie => (
      <MovieCard mv={movie} key={movie.id} />
    ))}
  </div>
);

SearchResults.propTypes = {
  moviesR: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = state => ({
  moviesR: state.movies,
});

export default connect(mapStateToProps)(SearchResults);
