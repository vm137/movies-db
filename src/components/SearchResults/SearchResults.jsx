// @flow

import React from 'react';
import MovieCard from '../MovieCard';
import './style.scss';

type Props = {
  moviesR: Object
}

const SearchResults = ({ moviesR }: Props) => (
  <div className="results-wrapper">
    {(moviesR).map(movie => (
      <MovieCard movie={movie} key={movie.id} />
    ))}
  </div>
);

export default SearchResults;
