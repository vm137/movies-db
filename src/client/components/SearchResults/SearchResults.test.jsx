import { shallow, render } from 'enzyme/build';
import React from 'react';
import testMovies from './__fixtures__/movies.json';
import SearchResults from './SearchResults';

const onClick = jest.fn();

describe('<SearchResults />', () => {
  it('renders matching snapshot', () => {
    const wrapper = shallow(<SearchResults
      movies={testMovies}
      onClick={onClick}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('check for .movie-card-wrapper', () => {
    const wrapper = render(<SearchResults
      movies={testMovies}
      onClick={onClick}
    />);
    expect(wrapper.find('.movie-card-wrapper').length).toBe(2);
  });
});
