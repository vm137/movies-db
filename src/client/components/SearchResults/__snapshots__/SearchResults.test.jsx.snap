// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> renders matching snapshot 1`] = `ShallowWrapper {}`;
